src/core/domain-services/crux-gateway.ts

```diff
@@ -11,27 +11,26 @@ import {
 
 export class CertificateManager {
     public static make = async (idClaim: IGatewayIdentityClaim, messageId: string): Promise<IGatewayIdentityCertificate> => {
-        const privateKey = "12381ab829318742938647283cd462738462873642ef34abefcd123501827193"; // foo@123
         const payload = {
             messageId,
         };
-        const signedProof = new TokenSigner("ES256K", privateKey).sign(payload);
+        const signedProof = await idClaim.keyManager.signWebToken(payload);
         return {
                 claim: idClaim.cruxId.toString(),
                 messageId,
                 proof: signedProof,
+                senderPubKey: await idClaim.keyManager.getPubKey(),
         };
     }
     public static verify = (idClaim: any, certificate: IGatewayIdentityCertificate) => {
         try {
-            if (!idClaim || !certificate.proof || !certificate.messageId) {
+            if (!idClaim) {
                 return false;
             }
             const proof: any = decodeToken(certificate.proof).payload;
-            const publicKey = idClaim.keyManager.publicKey;
-            const verified = new TokenVerifier("ES256K", publicKey).verify(certificate.proof);
+            const verified = new TokenVerifier("ES256K", certificate.senderPubKey).verify(certificate.proof);
```

Comment:
You can't use `certificate.senderPubKey` to verify this. Otherwise sender can make you believe that he's any body. Sender can send `ankit@foo.crux` as claim, but sign using his own private key. In public key he will send his own public key.

So you'll have to take `trustedPublicKey` as separate input apart from `certificate`. 

Caller will derive `trustedPublicKey` from `claim.cruxId` somehow

Reply:
modified the code accordingly